src/Persistence_matrix/concept/FieldOperators.h

```diff
@@ -0,0 +1,183 @@
+ * @ref Gudhi::persistence_fields::Multi_field_operators_with_small_characteristics.
+ */
+class FieldOperators 
+{
+ public:
+  using element_type = unspecified;         /**< Type for the elements in the field. */
+  using characteristic_type = unspecified;  /**< Type for the field characteristic. */
+
+  /**
+   * @brief Default constructor. If a non-zero characteristic is given, initializes the field with it.
+   * The characteristic can later be changed again or initialized with @ref set_characteristic.
+   *
+   * @param characteristic Prime number corresponding to the desired characteristic of the field.
+   */
+  FieldOperators(characteristic_type characteristic = 0);
+
+  /**
+   * @brief Sets the characteristic of the field. Can eventually be omitted if the characteristic of the class
+   * is fixed.
+   * 
+   * @param characteristic Prime number corresponding to the desired characteristic of the field.
+   */
+  void set_characteristic(characteristic_type characteristic);
+  /**
+   * @brief Returns the current characteristic.
+   * 
+   * @return The value of the current characteristic.
+   */
+  characteristic_type get_characteristic() const;
+
+  /**
+   * @brief Returns the value of an integer in the field.
+   * That is the positive value of the integer modulo the current characteristic.
+   * 
+   * @tparam Integer_type A native integer type: int, unsigned int, long int, bool, etc.
+   * @param e Integer to return the value from.
+   * @return @p e modulo the current characteristic, such that the result is positive.
+   */
+  template <typename Integer_type>
+  element_type get_value(Integer_type e) const;
+
+  /**
+   * @brief Returns the sum of two elements in the field.
+   * 
+   * @param e1 First element.
+   * @param e2 Second element.
+   * @return `(e1 + e2) % characteristic`, such that the result is positive.
+   */
+  element_type add(element_type e1, element_type e2) const;
+
+  // /**
+  //  * @brief Returns the substraction in the field of the first element by the second element.
+  //  * 
+  //  * @param e1 First element.
+  //  * @param e2 Second element.
+  //  * @return `(e1 - e2) % characteristic`, such that the result is positive.
+  //  */
+  // element_type substract(element_type e1, element_type e2) const;
+
+  /**
+   * @brief Returns the multiplication of two elements in the field.
+   * 
+   * @param e1 First element.
+   * @param e2 Second element.
+   * @return `(e1 * e2) % characteristic`, such that the result is positive.
+   */
+  element_type multiply(element_type e1, element_type e2) const;
+
+  /**
+   * @brief Multiplies the first element with the second one and adds the third one. Returns the result in the field.
+   * 
+   * @param e First element.
+   * @param m Second element.
+   * @param a Third element.
+   * @return `(e * m + a) % characteristic`, such that the result is positive.
+   */
+  element_type multiply_and_add(element_type e, element_type m, element_type a) const;
```

Comment:
I use 
```
mpz_mul(e.get_mpz_t(), e.get_mpz_t(), m.get_mpz_t());
mpz_add(e.get_mpz_t(), e.get_mpz_t(), a.get_mpz_t());
```
I don't know what happens behind these methods. I just assumed that because gmp provides those methods, that they were optimized. Or should I use `a += b * c` instead?

Reply:
I wasn't talking about the addition/multiplication itself, but about copies. If I write `x = add(x, y)`, I am going to allocate at least one extra object, while if I write `add_second_to_first(x, y)`, I can modify `x` in place and not return anything (same thing you do for your columns). So if a common use of multiply_and_add is `x=multiply_and_add(x,y,z)`, it may be interesting to have `void multiply_and_add(element_type&,element_type const&,element_type const&);`.

Reply:
Oh, you are right!

Reply:
I added inplace operations and replaced the old ones with them.